song/song_server.py

```diff
@@ -122,3 +133,72 @@ def _send_init_to_display(self):
         self.display_client.send_message(settings.DISPLAY_INIT_ADDRESS,
                                          pickle.dumps(self.song_machine.parser.categories, protocol=2)
                                          )
+
+class Tonality:
+    '''
+    idea of tonality counter combined with rules for FX chains and Sample Triggering in
+    Ableton Live
+    '''
+
+    tonality_counter = Counter()
+    FX_KEY = 'Clean'
+    ctrl_val = 0
+    last_cats = []
+    last_value = 0
+    most_common = ''
+    tonality_lock = False
+
+    category_to_chain = {
+        # every category points to a FX Chain and a ctrl value
+        'praise': ['Delay', 5],
+        'lecture': ['Vocoder', 8],
+        'insinuation': ['FreqShift', -7],
+        'dissence': ['Distortion', -5],
+        'concession': ['Delay', 2]
```

Comment:
It would be nice to have these assignments in a settings file rather than in the class. But I think we discussed about it and decided to do this later if necessary, right?